packages/api-client-core/src/GadgetConnection.ts

```diff
@@ -448,3 +458,13 @@ export class GadgetConnection {
     }
   }
 }
+
+function processMaybeRelativeInput(input: RequestInfo | URL, endpoint: string): RequestInfo | URL {
+  if (typeof input != "string") return input;
+  if (input.startsWith("/") && !input.startsWith("//")) {
```

Comment:
I think that'd be a nicer API, but i was a little worried about it being over eager and prepending the host to altogether malformed strings, which would really confuse things. If someone accidentally puts `ttp://` as their URL or something, I don't want to build a URL like `https://default-host/ttp://`. We could try to detect these kinds of cases but I think it will be a little sketchy, so I thought that maybe just a rule like "relative paths start with slashes and that's the only time we'll add weird implicit changes" seems kinda nice

Reply:
Yea, likely not worth the extra work to accommodate it considering that the semantics are the same. I think it's fine as long as we document it.